src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateDao.java

```diff
@@ -1,16 +1,25 @@
 package uk.gov.pay.directdebit.mandate.dao;
 
+import org.skife.jdbi.v2.sqlobject.Bind;
 import org.skife.jdbi.v2.sqlobject.BindBean;
 import org.skife.jdbi.v2.sqlobject.GetGeneratedKeys;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
 import org.skife.jdbi.v2.sqlobject.SqlUpdate;
 import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper;
+import org.skife.jdbi.v2.sqlobject.customizers.SingleValueResult;
 import uk.gov.pay.directdebit.mandate.dao.mapper.MandateMapper;
 import uk.gov.pay.directdebit.mandate.model.Mandate;
 
+import java.util.Optional;
+
 @RegisterMapper(MandateMapper.class)
 public interface MandateDao {
 
     @SqlUpdate("INSERT INTO mandates(payer_id, external_id) VALUES (:payerId, :externalId)")
     @GetGeneratedKeys
     Long insert(@BindBean Mandate mandate);
+
+    @SqlQuery("SELECT * FROM mandates m JOIN payers p ON m.payer_id = p.id JOIN transactions t ON t.payment_request_id = p.payment_request_id WHERE t.id = :transactionId")
```

Comment:
Sometimes it's a bit hard to understand which columns are mapped to which fields when we use `*` with multiple joins, for example I needed to refactor https://github.com/alphagov/pay-direct-debit-connector/pull/104/commits/ee8b4e969649a2cb922eac588672a8c504daf002 to map it correctly.
Is it good idea to do it here as well?

Reply:
I think it's necessary to be explicit when two or more tables have similarly named columns (ie. amount is in transaction and in payment request). You can get away with * when columns are different. I reason about that by looking at the mapper usually. If you think we should refactor I'll do it in the next pr